### ISSUE TITLE
IOException: SERVICE_NOT_AVAILABLE | AUTHENTICATION_FAILED | MISSING_INSTANCEID_SERVICE

### ISSUE THREAD

**Issue by nostra13** [helpcrunch/android-sdk-demo#35]
Crashes from prod:
 - `java.io.IOException: SERVICE_NOT_AVAILABLE`
 - `java.io.IOException: MISSING_INSTANCEID_SERVICE`
 - `java.io.IOException: AUTHENTICATION_FAILED`

```
Fatal Exception: com.helpcrunch.library.ge.j: java.io.IOException: SERVICE_NOT_AVAILABLE
       at com.google.android.gms.tasks.zzu.getResult(zzu.java:15)
       at com.helpcrunch.library.t1.a$a.onComplete(a.java:1)
       at com.google.android.gms.tasks.zzj.run(zzj.java:4)
       at android.os.Handler.handleCallback(Handler.java:900)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:219)
       at android.app.ActivityThread.main(ActivityThread.java:8349)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
```

`implementation "com.helpcrunch:chat-sdk:3.0.8-rc.03-s.85-SNAPSHOT"`
Probably the crash is actual for 3.0.8 too. 

**Devices** (Huawei, Xiaomi, Samsung):
PPA-LX1, JNY-LX1, Redmi Note 8 Pro, Mate 20 lite, HONOR 20 PRO, Galaxy S10e, Galaxy A50

**Comment by TalbotGooday** [comment]
Seems like this is a Firebase Exception. I'm sure that you may encounter this error on the Huawei (because of missing Google services) but I have never encountered something like this on Xiaomi or Samsung. Are you using the Firebase?

**Comment by nostra13** [comment]
Yes, I do. Some devices has no Google Play Services so FCM doesn't work on them. As I see from stacktrace: FCM produces exception but HelpCrunch doesn't catch it. I think it should. Crashing an app is not quite good solution for the case of absence (or some internal error) of Google Play Services.

**Comment by TalbotGooday** [comment]
Yes, you are right. I found a place where this might be happening.  The fix will be asap.  The question is, how is this happening.

**Comment by RKSharma89** [comment]
Still facing the same issue with Helpcrunch sdk 3.1.3

**Comment by TalbotGooday** [comment]
@RKSharma89 Are you getting these errors while testing on an emulator or in production?

**Comment by RKSharma89** [comment]
We are getting this issue in production. Lots of devices(Huawei, Samsung, Google Pixel, Xiaomi) are having this crash.

**Comment by nostra13** [comment]
Still stable crashing if Google Play Services are missing on device 
```
Fatal Exception: com.google.android.gms.tasks.RuntimeExecutionException: java.io.IOException: MISSING_INSTANCEID_SERVICE
       at com.google.android.gms.tasks.zzu.getResult(zzu.java:15)
       at com.helpcrunch.library.t1.a$a.onComplete(a.java:1)
       at com.google.android.gms.tasks.zzj.run(zzj.java:4)
       at android.os.Handler.handleCallback(Handler.java:900)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:219)
       at android.app.ActivityThread.main(ActivityThread.java:8299)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1052)
```

implementation "com.helpcrunch:chat-sdk:3.1.3"

**Comment by nostra13** [comment]
@TalbotGooday Happens again on 3.1.7: 
```
Fatal Exception: pa.g: java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: MISSING_INSTANCEID_SERVICE
       at com.google.android.gms.tasks.zzu.getResult(zzu.java:15)
       at com.helpcrunch.library.f.l.a$b.onComplete(a.java:1)
       at com.google.android.gms.tasks.zzi.run(zzi.java:4)
       at com.google.android.gms.cloudmessaging.zzk.run$bridge(zzk.java:35)
       at android.os.Handler.handleCallback(Handler.java:900)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at com.google.android.gms.internal.tasks.zzb.dispatchMessage(zzb.java:6)
       at android.os.Looper.loop(Looper.java:219)
       at android.app.ActivityThread.main(ActivityThread.java:8347)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
```
and
```
Fatal Exception: com.google.android.gms.tasks.RuntimeExecutionException: java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: SERVICE_NOT_AVAILABLE
       at com.google.android.gms.tasks.zzu.getResult(zzu.java:15)
       at com.helpcrunch.library.f.l.a$b.onComplete(a.java:1)
       at com.google.android.gms.tasks.zzj.run(zzj.java:4)
       at com.google.android.gms.cloudmessaging.zzk.run$bridge(zzk.java:40)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:236)
       at android.app.ActivityThread.main(ActivityThread.java:8043)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:620)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1011)
```

`implementation "com.helpcrunch:chat-sdk:3.1.7"`

**Comment by TalbotGooday** [comment]
Oh, my god. Thanks for reporting guys. Trying to investigate this problem

**Comment by TalbotGooday** [comment]
@nostra13 Finally reproduced and fixed in 3.1.8
